### PR TITLE
[SYCL-MLIR][NFC] Refactor Polygeist to add types definition

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Added `PolygeistTypes.[td,cpp,h]`.
Renamed `Ops.[h|cpp]` -> `PolygeistOps.[h|cpp]`, `Polygeist.h` -> `PolygeistDialect.h`, `Dialect.cpp` -> `PolygeistDialect.cpp`.

### CODE REVIEW

File: mlir-sycl/lib/Dialect/SYCL/IR/CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,7 @@ add_mlir_dialect_library(MLIRSYCLDialect
   SYCLTypes.cpp
 
   ADDITIONAL_HEADER_DIRS
-  ${MLIR_SYCL_SOURCE_DIR}/mlir/Dialect/SYCL
```
Comment (whitneywhtsang) [inline mlir-sycl/lib/Dialect/SYCL/IR/CMakeLists.txt:41]: The original path doesn't exist.

File: polygeist/lib/Dialect/Polygeist/IR/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,10 @@
 add_mlir_dialect_library(MLIRPolygeistDialect
-Dialect.cpp
-Ops.cpp
+PolygeistDialect.cpp
+PolygeistOps.cpp
+PolygeistTypes.cpp
 
 ADDITIONAL_HEADER_DIRS
-${PROJECT_SOURCE_DIR}/include/polygeist
```
Comment (whitneywhtsang) [inline polygeist/lib/Dialect/Polygeist/IR/CMakeLists.txt:6]: The original path doesn't exist.

File: polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistDialect.h [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 // This file is licensed under the Apache License v2.0 with LLVM Exceptions.
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef MLIR_DIALECT_POLYGEIST_IR_POLYGEIST_H_
-#define MLIR_DIALECT_POLYGEIST_IR_POLYGEIST_H_
+#ifndef MLIR_DIALECT_POLYGEIST_IR_POLYGEISTDIALECT_H_
+#define MLIR_DIALECT_POLYGEIST_IR_POLYGEISTDIALECT_H_
```
Comment (victor-eds) [inline polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistDialect.h:10]: I'd rather have the trailing `_` removed (same for all cases) as we did in the `sycl` dialect.,

File: polygeist/include/mlir/Dialect/Polygeist/IR/CMakeLists.txt [hunk 4]
```diff
@@ -1,2 +1,4 @@
 add_mlir_dialect(PolygeistOps polygeist)
-add_mlir_doc(PolygeistOps PolygeistOps Dialects/ -gen-dialect-doc -dialect=polygeist)
+add_mlir_doc(PolygeistBase PolygeistBase Dialects/ -gen-dialect-doc)
+add_mlir_doc(PolygeistOps PolygeistOps Dialects/ -gen-op-doc)
+add_mlir_doc(PolygeistTypes PolygeistTypes Dialects/ -gen-typedef-doc)
```
Comment (victor-eds) [inline polygeist/include/mlir/Dialect/Polygeist/IR/CMakeLists.txt:4]: Is the documentation being generated correctly? (should be target `polygeist-doc`)

File: polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistDialect.h (hunk 3 above)
Comment (whitneywhtsang) [reply polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistDialect.h]: done.

**Review by victor-eds — APPROVED** [review]
LGTM. I propose building `mlir-doc` as part of CI, though (different PR)

File: polygeist/include/mlir/Dialect/Polygeist/IR/CMakeLists.txt (hunk 4 above)
Comment (whitneywhtsang) [reply polygeist/include/mlir/Dialect/Polygeist/IR/CMakeLists.txt:4]: The documentation is now being generated correctly.

**Comment by whitneywhtsang** [comment]
> LGTM. I propose building `mlir-doc` as part of CI, though (different PR)

https://github.com/intel/llvm/pull/9283

**Comment by whitneywhtsang** [comment]
The CI failure started from https://github.com/intel/llvm/pull/9265.
```
Timed Out Tests (1):
  SYCL :: SubGroup/scan_spirv13.cpp
```

**Pull request merged**